src/wp-includes/author-template.php

```diff
@@ -437,12 +437,29 @@ function wp_list_authors( $args = '' ) {
 
 	$query_args           = wp_array_slice_assoc( $args, array( 'orderby', 'order', 'number', 'exclude', 'include' ) );
 	$query_args['fields'] = 'ids';
-	$authors              = get_users( $query_args );
 
 	$author_count = array();
-	foreach ( (array) $wpdb->get_results( "SELECT DISTINCT post_author, COUNT(ID) AS count FROM $wpdb->posts WHERE " . get_private_posts_cap_sql( 'post' ) . ' GROUP BY post_author' ) as $row ) {
+	$rows = $wpdb->get_results( "SELECT DISTINCT post_author, COUNT(ID) AS count FROM $wpdb->posts WHERE post_type = 'post' AND " . get_private_posts_cap_sql( 'post' ) . ' GROUP BY post_author' );
+	foreach ( $rows as $row ) {
 		$author_count[ $row->post_author ] = $row->count;
 	}
+
+	if ( $query_args['hide_empty'] ) {
```

Comment:
This is causing the tests to fail:

```
1) Tests_User_ListAuthors::test_wp_list_authors_default
Undefined index: hide_empty
```

There's no default value for the new argument.

Reply:
As I can see is set with a default at https://github.com/Mte90/wordpress-develop/blob/patch-2/src/wp-includes/author-template.php#L423

Reply:
That's for `$args`, not `$query_args`.

Reply:
You are right the refreshed patch had some obsolete code